app/controllers/icons_controller.rb

```diff
@@ -45,8 +45,7 @@ def show
     @page_title = @icon.keyword
     if params[:view] == 'posts'
       arel = Post.arel_table
-      post_ids = Reply.where(icon_id: @icon.id).select(:post_id).distinct.pluck(:post_id)
-      where_calc = arel[:icon_id].eq(@icon.id).or(arel[:id].in(post_ids))
+      where_calc = arel[:id].in(Reply.where(icon_id: @icon.id).select(:post_id).distinct.pluck(:post_id))
```

Comment:
We can leave this if you want but it's now really just `post_ids = Reply.where(icon_id: @icon.id).select(:post_id).distinct.pluck(:post_id)` and then `Post.where(id: post_ids)`, we don't need any of this arel stuff any more.